utility/pigpio/RF24_arch_config.h

```diff
@@ -43,6 +43,9 @@
     #endif
 #endif
 
+#undef RF24_SPI_SPEED
+#define RF24_SPI_SPEED 1000000
+    
```

Comment:
I think this is already defined in RF24_config.h (prior to `#include "utility/includes.h"`). 

https://github.com/nRF24/RF24/blob/0054260784f2ac116eed6ef9d9193ba133f3edcc/RF24_config.h#L42-L45

I only mention this because I think the `#undef` might override any values passed to `make` (like `make -DRF24_SPI_SPEED=4000000`) or CMake (which could also be done using `cmake .. -DRF24_SPI_SPEED=4000000; make ...`). Better to have 1 point of declaration, yes?

Reply:
I wanted to override the default of 10,000,000 with 1,000,000 specifically for pigpio since we know there are issues with higher spi speeds. How else to achieve that? Include a define in the build files if pigpio is selected?

Reply:
oh, I didn't count the zeroes. I guess we could move the define in RF24_config to after the `#include "utility/includes.h"` and use the `RF24_PIGPIO` macro to adjust that. like so:
```c
/** @brief The default SPI speed (in Hz) */ 
 #ifndef RF24_SPI_SPEED
    #ifdef RF24_PIGPIO
         #define RF24_SPI_SPEED 1000000
    #else
         #define RF24_SPI_SPEED 10000000 
    #endif
 #endif 
```

Reply:
Don't think that'll work. It needs to be defined globally and before includes.h which includes the SPI.h file.

How about something like the following in RF24_arch_config.h?:
```
#if RF24_SPI_SPEED > 10000000
   #undef RF24_SPI_SPEED
   #define RF24_SPI_SPEED 1000000
#endif
```
This allows users to define it, but we just put a cap on the maximum value.

Reply:
That's also acceptable, though I think you meant `RF24_SPI_SPEED > 1MHz` (or `>= 10MHz`)

> Don't think that'll work.

I think it should. @Avamander  Am I assuming correctly that macros are only evaluated at compile time after including all necessary files?

Furthermore, the use of `RF24_SPI_SPEED` as default values in spi.h shouldn't be needed as RF24 class will always specify the SPI speed when init-ing the SPI obj.

Reply:
Updated. Its getting weird that you are starting to know more about this library than me... 👍